src/sentry/search/snuba/backend.py

```diff
@@ -317,16 +318,45 @@ def _query(self, projects, retention_window_start, group_queryset, environments,
                 ),
             }).build(group_queryset, search_filters)
         else:
+
             group_queryset = QuerySetBuilder({
                 'first_release': QCallbackCondition(
-                    lambda version: Q(
-                        first_release__organization_id=projects[0].organization_id,
-                        first_release__version=version,
+                    lambda release_version: Q(
+                        # if no specific environments are supplied, we choose any groups that has a specific first release
+                        # that matches the given release_version, (agnostic of environment)
+                        Q(first_release__version=release_version) |
+                        # or choose any groups whose first occurrence in any environment and the latest release at
+                        # the time of the groups' occurrence matches the given release_version
+                        Q(groupenvironment__first_release__version=release_version),
+                        first_release__organization_id=projects[0].organization_id
                     ),
                 ),
                 'first_seen': ScalarCondition('first_seen'),
             }).build(group_queryset, search_filters)
 
+            if 'first_release' in search_filters:
```

Comment:
@markstory I've isolated the specialized query and queryset modifications only when first_release exists in search_filters